builder/azure/common/client/config.go

```diff
@@ -228,28 +233,32 @@ func (c Config) Validate(errs *packersdk.MultiError) {
 	}
 
 	errs = packersdk.MultiErrorAppend(errs, fmt.Errorf("No valid set of authentication values specified:\n"+
-		"  to use the Managed Identity of the current machine, do not specify any of the fields below\n"+
-		"  to use interactive user authentication, specify only subscription_id\n"+
+		"  to use the Managed Identity of the current machine, do not specify any of the fields below:\n"+
+		"  - client_secret\n"+
+		"  - client_jwt\n"+
+		"  - client_cert_path\n"+
+		"  - use_interactive_auth\n"+
+		"  - use_azure_cli_auth\n"+
+		"  to use interactive user authentication, specify only the following fields:\n"+
+		"  - subscription_id\n"+
+		"  - use_interactive_auth\n"+
 		"  to use an Azure Active Directory service principal, specify either:\n"+
 		"  - subscription_id, client_id and client_secret\n"+
 		"  - subscription_id, client_id and client_cert_path\n"+
 		"  - subscription_id, client_id and client_jwt."))
 }
 
 func (c Config) useDeviceLogin() bool {
-	return c.SubscriptionID != "" &&
-		c.ClientID == "" &&
-		c.ClientSecret == "" &&
-		c.ClientJWT == "" &&
-		c.ClientCertPath == ""
+	return c.UseInteractiveAuth
```

Comment:
Can you update the comment documentation here https://github.com/hashicorp/packer-plugin-azure/blob/d0b03117e54573a2ffbdf1df944bd661fdc4e6af/builder/azure/common/client/config.go#L169 in this file as well for the new behavior of useDeviceLogin